source/DEC_API.c

```diff
@@ -21,11 +21,15 @@
 #include "global.h"
 #include "DEC_API.h"
 
+static char dec_last = 0x00;
+
 void DEC_init(){
     LOG("rotencoder init\n");
```

Comment:
Please use a dedicated logging level e.g. "DEG_LOG()" or leave it out. I think its the only LOG() you used so you can leave it out.